core/arch/arm/plat-mediatek/core_bootcfg.c

```diff
@@ -0,0 +1,212 @@
+	MEMACCESS_AREA(TZSRAM_BASE, TZSRAM_SIZE),
+#endif
+	MEMACCESS_AREA(TZDRAM_BASE, TZDRAM_SIZE),
+};
+
+static struct memaccess_area nsec_shared[] = {
+	MEMACCESS_AREA(CFG_SHMEM_START, CFG_SHMEM_SIZE),
+};
+
+static bool _pbuf_intersects(struct memaccess_area *a, size_t alen,
+			     paddr_t pa, size_t size)
+{
+	size_t n;
+
+	for (n = 0; n < alen; n++)
+		if (core_is_buffer_intersect(pa, size, a[n].paddr, a[n].size))
+			return true;
+	return false;
+}
+#define pbuf_intersects(a, pa, size) \
+	_pbuf_intersects((a), ARRAY_SIZE(a), (pa), (size))
+
+static bool _pbuf_is_inside(struct memaccess_area *a, size_t alen,
+			    paddr_t pa, size_t size)
+{
+	size_t n;
+
+	for (n = 0; n < alen; n++)
+		if (core_is_buffer_inside(pa, size, a[n].paddr, a[n].size))
+			return true;
+	return false;
+}
+#define pbuf_is_inside(a, pa, size) \
+	_pbuf_is_inside((a), ARRAY_SIZE(a), (pa), (size))
+
+static bool pbuf_is_multipurpose(paddr_t paddr, size_t size)
+{
+	if (pbuf_intersects(secure_only, paddr, size))
+		return false;
+	if (pbuf_intersects(nsec_shared, paddr, size))
+		return false;
+
+	return pbuf_is_inside(ddr, paddr, size);
+}
+
+/*
+ * Wrapper for the platform specific pbuf_is() service.
+ */
+static bool pbuf_is(enum buf_is_attr attr, paddr_t paddr, size_t size)
+{
+	switch (attr) {
+	case CORE_MEM_SEC:
+		return pbuf_is_inside(secure_only, paddr, size);
+
+	case CORE_MEM_NON_SEC:
+		return pbuf_is_inside(nsec_shared, paddr, size);
+
+	case CORE_MEM_MULTPURPOSE:
+		return pbuf_is_multipurpose(paddr, size);
+
+	case CORE_MEM_EXTRAM:
+		return pbuf_is_inside(ddr, paddr, size);
+
+	default:
+		EMSG("unpexted request: attr=%X", attr);
+		return false;
+	}
+}
+
+/* platform specific memory layout provided to teecore */
+static struct map_area bootcfg_memory_map[] = {
+	{	/* teecore execution RAM */
+	 .type = MEM_AREA_TEE_RAM,
+	 .pa = CFG_TEE_RAM_START, .size = CFG_TEE_RAM_PH_SIZE,
+#ifdef CFG_WITH_PAGER
+	 .region_size = SMALL_PAGE_SIZE,
+#endif
+	 .cached = true, .secure = true, .rw = true, .exec = true,
+	 },
+
+	{	/* teecore TA load/exec RAM - Secure, exec user only! */
+	 .type = MEM_AREA_TA_RAM,
+	 .pa = CFG_TA_RAM_START, .size = CFG_TA_RAM_SIZE,
+	 .cached = true, .secure = true, .rw = true, .exec = false,
+	 },
+
+	{	/* teecore public RAM - NonSecure, non-exec. */
+	 .type = MEM_AREA_NSEC_SHM,
+	 .pa = CFG_SHMEM_START, .size = CFG_SHMEM_SIZE,
+	 .cached = true, .secure = false, .rw = true, .exec = false,
+	 },
+
+	{
+	 .type = MEM_AREA_IO_NSEC,
+	 .pa = DEVICE0_BASE, .size = DEVICE0_SIZE,
+	 .device = true, .secure = true, .rw = true,
+	 },
+	{
+	 .type = MEM_AREA_IO_SEC,
+	 .pa = DEVICE1_BASE, .size = DEVICE1_SIZE,
+	 .device = true, .secure = true, .rw = true,
+	 },
+	{.type = MEM_AREA_NOTYPE}
+};
+
+/*
+ * bootcfg_get_pbuf_is_handler - return the platform specific pbuf_is
+ */
+unsigned long bootcfg_get_pbuf_is_handler(void)
+{
+	return (unsigned long)pbuf_is;
+}
+
+/*
+ * This routine is called while MMU and core memory management are not init.
```

Comment:
I suggest:
"This routine is called when MMU and core memory management are not initialized."